# Chapter 4 - Problems

In [1]:
from PhysicsOneA import *
from PhysicsOneA.dependencies import *

c:\Users\victo\AppData\Local\Programs\Python\Python312\Lib\site-packages\uncertainties\core.py:1024: UserWarning: Using UFloat objects with std_dev==0 may give unexpected results.
  warn("Using UFloat objects with std_dev==0 may give unexpected results.")


# Exercise 1

In [2]:
# Just rank them.
# Point 1 is greatest velocity, Point 2 is least velocity, Point 3 is second greatest velocity
points = [["v1",3],["v2",1],["v3",2]]
# Just sort them, imma use bubble sort cuz im lazy
for n in range(len(points) - 1, 0, -1):
    swapped = False  
    for i in range(n):
        if points[i][1] > points[i + 1][1]:
            points[i], points[i + 1] = points[i + 1], points[i]
            swapped = True
    if not swapped:
        break

print(points)

[['v2', 1], ['v3', 2], ['v1', 3]]


So the correct answer is 
v2 < v3 < v1

Correct Answer: E v2 < v3 < v1

# Exercise 2

Use the slides or book for this one champ, correct answer E

# Exercise 3

a)

In [3]:
# For this exercise, we just gotta make a circular motion
r = 6370 * 1000 # cuz radius needs to be in meter
T = 60*60*24 # cuz it takes a day for the earth to rotate (needs to be in seconds)
c = CircularMotion(r=r, T=T)
print(c)

--- Circular Motion ---
Radius r: 6370000.00000 ± 0.00000 m
Period T: 86400.00000 ± 0.00000 s
v = 463.23947 ± 0.00000 m/s
centripetal acceleration = 0.03369 ± 0.00000 m/s²
tangential acceleration = N/A
Critical period for weightlessness: 5060.50 ± 0.00 seconds (~1.41 ± 0.00 hours)


So answer is 0.03369 m/s^2, because it's a person on the edge of the circle, so centripetal force!

b)

Now just read again. Critical period for weightlessness = 5060 seconds or 1.41 hours

# Exercise 4

In [4]:
angle = degree_to_radian(50) # radians
tof = 3.5 # seconds (time of flight)
proj = Projectile(theta=angle, time_of_flight=tof)
print(proj)

--- Projectile Info ---
Initial speed (v0): 22.4334+/-0 m/s
Launch angle (theta): 50.00+/-0°
Horizontal velocity (v0x): 14.4199+/-0 m/s
Vertical velocity (v0y): 17.1850+/-0 m/s
Initial height (y0): 0.0000+/-0 m
Gravity (g): 9.8200+/-0 m/s²

--- Derived Quantities ---
Time of flight (flat): 3.5000+/-0 s
Time of flight (realistic): 3.5000+/-0 s
Maximum height: 15.0369+/-0 m
Range: 50.4697+/-0 m


C:\Users\victo\Documents\DTU Kurser\Fysik 1\PhysicsOneA\src\PhysicsOneA\Kinematics2D.py:286: FutureWarning: AffineScalarFunc.__gt__() is deprecated. It will be removed in a future release.
  return max(t1, t2)


Initial speed is 14.42 m/s, cause it is thrown horizontally as seen on the illustration, so it's actually not the v0 variable, it's the v0x variable!

# Exercise 5

In [5]:
height = 30/100 # cause it needs to be meters
length = 10 # meters

proj = Projectile(max_height=height, range_=length)
print(proj)

--- Projectile Info ---
Initial speed (v0): 20.3812+/-0 m/s
Launch angle (theta): 6.84+/-0°
Horizontal velocity (v0x): 20.2361+/-0 m/s
Vertical velocity (v0y): 2.4273+/-0 m/s
Initial height (y0): 0.0000+/-0 m
Gravity (g): 9.8200+/-0 m/s²

--- Derived Quantities ---
Time of flight (flat): 0.4944+/-0 s
Time of flight (realistic): 0.4944+/-0 s
Maximum height: 0.3000+/-0 m
Range: 10.0041+/-0 m


C:\Users\victo\Documents\DTU Kurser\Fysik 1\PhysicsOneA\src\PhysicsOneA\Kinematics2D.py:183: FutureWarning: AffineScalarFunc.__abs__() is deprecated. It will be removed in a future release.
  err = abs(lhs - rhs)
C:\Users\victo\Documents\DTU Kurser\Fysik 1\PhysicsOneA\src\PhysicsOneA\Kinematics2D.py:184: FutureWarning: AffineScalarFunc.__lt__() is deprecated. It will be removed in a future release.
  if err < min_error:


So initial speed = 20.3812 m/s

initial angle = 6.84 degrees

duration of flight = 0.4944 s

easy innit?

# Exercise 6

So another projectile xD

In [7]:
p = Projectile(y0=3, max_height=5.5, range_=2.8, impact_height=0)
print(p)

--- Projectile Info ---
Initial speed (v0): 7.1831+/-0 m/s
Launch angle (theta): 77.29+/-0°
Horizontal velocity (v0x): 1.5802+/-0 m/s
Vertical velocity (v0y): 7.0071+/-0 m/s
Initial height (y0): 3.0000+/-0 m
Gravity (g): 9.8200+/-0 m/s²

--- Derived Quantities ---
Time of flight (flat): 1.4271+/-0 s
Time of flight (realistic): 1.7719+/-0 s
Maximum height: 5.5000+/-0 m
Range: 2.2551+/-0 m


Just read this